### PR TITLE
Feat/hng 26 currency formater component

### LINKED ISSUE #26: [FEATURE] 🗺️ User Dashboard & Products Pages > Dashboard > implement a reusable  Currency Formatter component
- https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=4-49804&t=ybfmYNMB6xh8HxXT-1

### **Description**
Develop a reusable Currency Formatter component. This component will be used to format and display currency values across the website. The Currency Formatter should be accurate, versatile, and adhere to international currency standards. It should include the following features:

Acceptance Criteria

The Currency Formatter component is implemented according to the design.
The component should correctly format currency values for different currencies.
Ensure accessibility standards are met, including appropriate labeling for screen readers.
The component should handle edge cases such as large values, negative numbers, and zero values.
Requirements

The component should receive the following props:

1. amount: The numerical value to be formatted.
2. currency: The currency code (e.g., "USD", "EUR", "JPY").
3. showSymbol: A boolean to determine whether to display the currency symbol.

Tasks

- [ ] Develop the Currency Formatter component based on the provided requirements.
- [ ] Implement currency code handling.
- [ ] Ensure the component renders correctly with various currency values.
- [ ] Check that the component handles different currency codes.
- [ ] Test the component with edge cases like large values, negative numbers, and zero.
- [ ] Ensure proper handling of edge cases.
- [ ] test for cases where value is null, symbol is null, or value is a string
- [ ] Test the component for accuracy and access
Data Card (1)

### PR DESCRIPTION
**Pull request by Amdragz**
# Description
This PR addresses the issue by implementing a Currency Converter component which allows users to convert an amount from USD to various currencies and display the converted amount with the appropriate currency symbol.

**Closes #issue_number_here**
Closes #26

# Changes proposed

## What were you told to do?
 Implement a reusable Currency Formatter component 
The task was to develop a Currency Converter component that takes an amount, showSymbol and a currency code as props, and converts the amount according to the currency and then render the amount and if showSymbol is passed render the currency symbol on the screen

## What did you do?
Implemented the Currency Converter component.
Added fetchRates function to retrieve conversion rates from the API.
Integrated convertCurrencyFromUSD function to perform currency conversion.
Managed component states for loading, error handling, and displaying the result.
Added unit tests to cover the  successful data fetching, and conversion calculations.

I created the component under the common component, i created the currency converter component and two unit test files for it, i implemented the currency converter component by fetching the exchange rate for the currencies from a free API site and use the rate to carryout a simple calculation in a convertCurrencyyFromUSD function. Then i wrote test to test my functions and what would happen for different edge cases, i did not change any files and i did basic styling with tailwind css

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/Videos
[2024-07-20 19-57-33.zip](https://github.com/user-attachments/files/16322078/2024-07-20.19-57-33.zip)

### CODE REVIEW

File: src/components/common/CurrencyConverter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+      {
+        method: "GET",
+      },
+    );
+
+    if (!response.ok) {
+      throw new Error("Failed to fetch currency rates");
+    }
+    const data = await response.json();
+
```
Comment (oreoluwa212) [inline src/components/common/CurrencyConverter/index.tsx]: This whole block, remove it, thank you

Logging to the console is generally prohibited in this stage. Thanks again

So, I see potential here but please you remember stage 2 bah, static just static,  dont overdo. Thanks 

Waiting for your kind corrections. I need a video of you testing your component

Add screenshots to your pnpm test and lint test. Thanksssss

**Review by oreoluwa212 — CHANGES_REQUESTED** [review]
...thank....you

**Comment by Amdragz** [comment]
I have refactored the code to remove any instance of console.logs, i made the fetchRate function return some mock conversion rates instead of logging errors 
[HNG-#26-test.zip](https://github.com/user-attachments/files/16323896/HNG-.26-test.zip)
[HNG-test-26-take-2 2024-07-21 113607.zip](https://github.com/user-attachments/files/16323912/HNG-test-26-take-2.2024-07-21.113607.zip)

**Comment by Amdragz** [comment]
Please merge my branch I have reviwed and made the requested changes
